migration/idempotent/011-maintenance.sql

```diff
@@ -444,12 +444,25 @@ COMMENT ON PROCEDURE _prom_catalog.execute_data_retention_policy(boolean)
 IS 'drops old data according to the data retention policy. This procedure should be run regularly in a cron job';
 GRANT EXECUTE ON PROCEDURE _prom_catalog.execute_data_retention_policy(boolean) TO prom_maintenance;
 
+CREATE OR REPLACE PROCEDURE prom_api.execute_maintenance(log_verbose boolean = false)
+AS $$
+BEGIN
+    CALL prom_api.execute_maintenance('metrics', 'compression', log_verbose);
+    CALL prom_api.execute_maintenance('metrics', 'retention', log_verbose);
```

Comment:
if there's ever a case where an old uncompressed chunk is ready to be dropped by retention, better to drop it first rather than compress and then drop? I wonder if the compression job is smart enough to not compress a chunk that's older than the retention window

Reply:
Good catch! The legacy order ran both retentions first.